transport/src/main/java/io/netty/channel/FileRegion.java

```diff
@@ -84,9 +86,86 @@
 
+    /**
+     * Transfers the content of this file region to the specified channel.<p>
+     * This method does not modify {@link #transferIndex()}.
+     *
+     * @param target    the destination of the transfer
+     * @param position  the relative offset of the file where the transfer
+     *                  begins from.  For example, <tt>0</tt> will make the
+     *                  transfer start from {@link #position()}th byte and
+     *                  <tt>{@link #count()} - 1</tt> will make the last
+     *                  byte of the region transferred.
+     * @param length    the maximum number of bytes to transfer
+     */
+    long transferBytesTo(WritableByteChannel target, long position, long length) throws IOException;
+
+    /**
+     * Transfers the content of this file region to the specified channel.
+     *
+     * @param target    the destination of the transfer
+     * @param length    the maximum number of bytes to transfer
+     */
+    long transferBytesTo(WritableByteChannel target, long length) throws IOException;
+
+    /**
+     * Returns the {@code transferIndex} of this file region.
+     */
+    long transferIndex();
+
+    /**
+     * Sets the {@code transferIndex} of file region.
+     *
+     * @throws IndexOutOfBoundsException
+     *             if the specified {@code transferIndex} is less than {@code 0} or greater than {@link #count()}
+     */
+    FileRegion transferIndex(long index);
+
+    /**
+     * Returns {@code true} if and only if {@link #transferableBytes()} is greater than {@code 0}.
+     */
+    boolean isTransferable();
+
+    /**
+     * Returns the number of readable bytes which is equal to {@code (count - transferIndex)}.
+     */
+    long transferableBytes();
+
+    /**
+     * Returns a slice of this file region's sub-region. This method does not modify {@code transferIndex} of this
+     * buffer.
+     * <p>
+     * Also be aware that this method will NOT call {@link #retain()} and so the reference count will NOT be increased.
+     */
+    FileRegion slice(long index, long length);
+
+    /**
+     * Returns a new slice of this file region's sub-region starting at the current {@code transferIndex} and increases
+     * the {@code transferIndex} by the size of the new slice (= {@code length}).
+     * <p>
+     * Also be aware that this method will NOT call {@link #retain()} and so the reference count will NOT be increased.
+     *
+     * @param length    the size of the new slice
+     */
+    FileRegion transferSlice(long length);
+
+    /**
+     * Return the underlying file region instance if this file region is a wrapper of another file region.
+     *
+     * @return {@code null} if this file region is not a wrapper
+     */
+    FileRegion unwrap();
+
+    /**
+     * Return the underlying file channel instance of this file region.
+     */
+    FileChannel channel() throws IOException;
```

Comment:
see above also I think this should be not part of the `FileRegion` interface at all as the FileRegion should not be specific to `FileChannel`.